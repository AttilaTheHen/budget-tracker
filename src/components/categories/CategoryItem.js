import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { remove } from './actions';
import Expenses from './expenses/Expenses';

export class CategoryItem extends Component {
  static propTypes = {
    category: PropTypes.object.isRequired,
    onEdit: PropTypes.func.isRequired,
    remove: PropTypes.func.isRequired
  };
  
  render() {
    const { category, onEdit, remove } = this.props;
    
    return (
      <section>
        <p>{category.name}</p>
        <p>{category.budget}</p>
        <button name="edit" onClick={onEdit}>Edit</button>
        <button name="delete" onClick={() => remove(category.key)}>Delete</button>
        <Expenses
          categoryId={category.key}
        />
      </section>
    );
  }
}
 
export default connect(
  null,
  { remove }
)(CategoryItem);