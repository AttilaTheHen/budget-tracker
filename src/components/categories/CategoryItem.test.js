import { CategoryItem } from './CategoryItem';
import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

describe('Category item display', () => {
  it('renders item display, then edit', () => {
    const handleEdit = jest.fn();
    const remove = jest.fn();

    const wrapper = shallow(<CategoryItem
      category={{ name: 'name', budget: 250 }}
      onEdit={handleEdit}
      remove={remove}
    />);

    expect(toJSON(wrapper)).toMatchSnapshot();

    wrapper.find('button[name="edit"]').simulate('click');
    const editCalls = handleEdit.mock.calls;
    expect(editCalls.length).toBe(1);

    wrapper.find('button[name="delete"]').simulate('click');
    const deleteCalls = remove.mock.calls;
    expect(deleteCalls.length).toBe(1);
  });
});