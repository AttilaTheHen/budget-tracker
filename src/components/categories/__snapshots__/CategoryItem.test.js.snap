// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category item display renders item display, then edit 1`] = `
<section>
  <p>
    name
  </p>
  <p>
    250
  </p>
  <button
    name="edit"
    onClick={[MockFunction]}
  >
    Edit
  </button>
  <button
    name="delete"
    onClick={[Function]}
  >
    Delete
  </button>
</section>
`;
